### ISSUE TITLE
Document that RequestMapping over a FeignClient interface should not be used

### ISSUE THREAD

**Issue by cbornet** [spring-cloud/spring-cloud-netflix#1079]
It seems that Spring-MVC interprets it as a controller.
If you use an Hystrix fallback inheriting from the interface, then it even fails at startup saying that there are two controllers for the same mapping.

**Comment by cbornet** [comment]
See https://github.com/spring-cloud/spring-cloud-netflix/issues/466

**Comment by dsyer** [comment]
Clarification: ut's only a problem if you have both a `@FeignClient` and a `@Controller` with the same type-level annotation (e.g. if they share an interface). Since this makes very little sense for a real application, it's not super urgent to do anything on our side (and it would take a change in Spring Framework anyway) to support it. Docmuentation is definitely a good idea if anyone wants to propose a change.

**Comment by cbornet** [comment]
Note : in a spring-boot app, it crashes if you have both a @FeignClient and a FeignHystrixCallback implementing the interface because Spring-MVC complains about ambiguous path mapping.

**Comment by dsyer** [comment]
Right, yes, that as well. That's worth documenting as well: you can't use `fallback=` in `@FeignClient` if there is a `@RequestMapping` on the interface.

**Comment by cbornet** [comment]
It also gets scanned by springfox and probably other tools. Lots of worries :smile: ...

**Comment by heymysweetheart** [comment]
Yes, when I used swagger and there are two controller and feignclient but accidently they have the same  RequestMapping, then there are `IllegalStateException: Ambiguous mapping`, and the project failed starting.
